src/threads/table.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { dynamodb, toDBItem } from '../dynamodb';
+import { ForumTable } from './table';
+
+const putItem = (threadId: string, timestamp: number, data: Record<string, unknown>) => dynamodb.putItem({
+  TableName: 'forumTable',
+  Item: toDBItem({ threadId, timestamp, ...data }),
+});
+const getAll = (threadId: string) => dynamodb.query({
+  TableName: 'forumTable',
+  ExpressionAttributeValues: { ':tid': { S: threadId } },
+  KeyConditionExpression: 'threadId = :tid',
+});
+const deleteAllItems = async (threadId: string) => {
+  const result = await getAll(threadId);
+  await Promise.all((result.Items || []).map(item => dynamodb.deleteItem({ TableName: 'forumTable', Key: item })));
+};
+
+describe('Forum table', () => {
+  const forumTable = new ForumTable(dynamodb);
+  const consoleErrorStub = jest.spyOn(global.console, 'error');
+  consoleErrorStub.mockImplementation(() => {});
+
+  describe('getThreadEvents()', () => {
+    const participantId = 'participantId';
+    const itemId = 'itemId';
+    const threadId = `THREAD#${participantId}#${itemId}`;
+
+    afterEach(async () => {
+      await deleteAllItems(threadId);
+    });
+
+    it('should fail at decode step', async () => {
+      await putItem(threadId, 1, { type: 'unknown_type' });
+      await expect(forumTable.getThreadEvents(participantId, itemId)).rejects.toBeInstanceOf(Error);
```

Comment:
As discussed, the service has to be robust, not returning errors if some data does not match the current format. (may be an older message format that we were unable to remove as cleanup may be hard to do in such dbs)

Reply:
Currently we are the only ones to add data in the db with formats we know. To me, we can start with strict decoders, and later on when we add properties to any object, we'll add each property as optional and decode them using `D.partial(...)` decoder.
Or is it so unpredictable that we can't even be sure that added data will have the same shape in output as input?

Reply:
In such a db, you cannot do a single assumption on the data you get. Unlike a relation db, if a tester/dev enters dumb data, it may completely differ from your expected schema and will not be cleaned with any migration. 
It is not unexpected in the sense the system may create random data by itself, but a old transient bug, a test, a old conflicting feature,... may have added info which does not meet what you expect, and you do not have good way to fix these data. The best strategy is such db is just to ignore data silently which is in a expected format that you are just not affected by them.

Of course, it may be done later, but failing on invalid data looks dangerous as it may cause your app to suddenly stop working for obscure reasons.